File: Code_For_Test/sample_code.php
<?php
/**
 * Sample PHP file for testing Interview Code Lens
 * This file demonstrates various PHP features
 */

/**
 * Calculate the sum of an array of numbers
 */
function calculateSum(array $numbers): int {
    return array_sum($numbers);
}

/**
 * Process array data and return formatted string
 */
function processData(array $data): string {
    return json_encode($data, JSON_PRETTY_PRINT);
}

/**
 * Fetch content from URL
 */
function fetchUrl(string $url): string {
    return file_get_contents($url);
}

/**
 * A class to process various types of data
 */
class DataProcessor {
    private $config;
    
    /**
     * Constructor
     */
    public function __construct(array $config) {
        $this->config = $config;
    }
    
    /**
     * Process data based on configuration
     */
    public function process(): string {
        return $this->config['output'] ?? 'default';
    }
    
    /**
     * Get configuration
     */
    public function getConfig(): array {
        return $this->config;
    }
    
    /**
     * Static method to create default instance
     */
    public static function createDefault(): DataProcessor {
        return new self(['output' => 'default']);
    }
}

// Example usage
if (php_sapi_name() === 'cli') {
    $numbers = [1, 2, 3, 4, 5];
    $result = calculateSum($numbers);
    echo "Sum: $result\n";
}
?>

File: Code_For_Test/sample_code.kt
/**
 * Sample Kotlin file for testing Interview Code Lens
 * This file demonstrates various Kotlin features
 */
package sample

import kotlinx.coroutines.*
import java.net.URL
import com.google.gson.GsonBuilder

/**
 * Calculate the sum of a list of numbers
 */
fun calculateSum(numbers: List<Int>): Int {
    return numbers.sum()
}

/**
 * Process map data and return formatted string
 */
fun processData(data: Map<String, Any>): String {
    val gson = GsonBuilder().setPrettyPrinting().create()
    return gson.toJson(data)
}

/**
 * Fetch content from URL using coroutines
 */
suspend fun fetchUrl(url: String): String {
    return withContext(Dispatchers.IO) {
        URL(url).readText()
    }
}

/**
 * A class to process various types of data
 */
class DataProcessor(private val config: Map<String, Any>) {
    
    /**
     * Process data based on configuration
     */
    fun process(): String {
        return config["output"] as? String ?: "default"
    }
    
    /**
     * Get configuration
     */
    fun getConfig(): Map<String, Any> {
        return config.toMap()
    }
    
    companion object {
        /**
         * Create default instance
         */
        fun createDefault(): DataProcessor {
            return DataProcessor(mapOf("output" to "default"))
        }
    }
}

/**
 * Main function for testing
 */
fun main() {
    val numbers = listOf(1, 2, 3, 4, 5)
    val result = calculateSum(numbers)
    println("Sum: $result")
}
